packages/react-router-dev/vite/plugin.ts

```diff
@@ -1033,6 +1033,7 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = (_config) => {
                   plugin !== null &&
                   "name" in plugin &&
                   plugin.name !== "react-router" &&
+                  plugin.name !== "react-router-route-exports" &&
```

Comment:
@markdalgleish as I was working on this, I realized that the child compiler really exists to transform the routes with all plugins _prior_ to the Remix/RR plugins. It just so happens that all of our transforms only target client code (`ssr: false`) so (1) the early returns in each transform (`if (options?.ssr) return`) combined with (2) setting `ssr: true` when using the child compiler to determine route exports, just happened to work.

But now that we have a transform that needs to run for both (the props HOC transform), that stopped working. I added `react-router-route-exports` to this list, but what if we just add all the plugins defined here to this list? we could even define the plugins separately. That way we'll still get all external plugins set in the child compiler.

Reply:
Yeah, that makes sense. Would it be too aggressive to simply filter out `react-router` and anything starting with `react-router-`? Alternatively, we could annotate all of our plugins with an extra property that we detect (e.g. we check for `plugin.__reactRouterPlugin` or something similar). That way we don't have to maintain a list.